File: pkg/models/media.go
package models

import "time"

type MediaType string
type MediaStatus string

const (
	MediaTypePhoto MediaType = "photo"
	MediaTypeAudio MediaType = "audio"
	MediaTypeGif   MediaType = "gif"
	MediaTypeVideo MediaType = "video"

	MediaStatusDraft      = "draft"
	MediaStatusProcessing = "processing"
	MediaStatusCompleted  = "completed"
)

// Media is the representation of a media in the database
type Media struct {
	ID         string      `json:"id" bson:"_id"`
	InternalID string      `json:"-" bson:"internal_id"`
	CreatedAt  time.Time   `json:"created_at" bson:"created_at"`
	Title      string      `json:"title" bson:"title"`
	Status     MediaStatus `json:"status" bson:"status"`
	Type       MediaType   `json:"type" bson:"type"`
	Tags       []string    `json:"tags" bson:"tags"`
	UserID     string      `json:"user_id" bson:"user_id"`
}

// PostMediaResponse is the response shown when a media is created or updated
type PostMediaResponse struct {
	Media
}

File: frontend/src/lib/api/media.ts
import type { Axios } from "axios"
import { NewApiClient } from "./apiClient"
import type { AxiosInstance } from "axios"

const baseURL = '/media'
export class MediaAPI {
    private axiosInstance: AxiosInstance

    constructor(axiosInstance: AxiosInstance) {
        this.axiosInstance = axiosInstance
    }

    ReqUpload(name: string, file: File) {
        return new Promise<MediaUploadResponse>((resolve, reject) => {
            return this.axiosInstance.post(
                `${baseURL}?title=${encodeURIComponent(name)}`,
                file)
                .then(repsonse => {
                    const data: MediaUploadResponse = repsonse.data
                    resolve(data)
                })
                .catch(err => {
                    reject(err.response.data.error)
                })
        })
    }
}

export interface MediaUploadResponse extends Media {

}

export enum MediaStatus {
    Processing = "processing",
    Completed = "completed",
    Draft = "draft"
}

export enum MediaType {
    Image = "image",
    Video = "video",
    Audio = "audio",
    Document = "document",
    Other = "other"
}


export interface Media {
    id: string;
    created_at: Date;
    title: string;
    status: MediaStatus;
    type: MediaType;
    tags: string[];
    user_id: string;
}